### ISSUE TITLE
RMSD weights not applied to groupselections

### ISSUE THREAD

**Issue by lilyminium** [MDAnalysis/mdanalysis#2429]
**Expected behavior**

I expect that when `weights='mass'` is passed to `MDAnalysis.analysis.rms.RMSD`, it means that an array of masses is created from the atoms I have selected, for ``groupselections`` as well as ``select``.

**Actual behavior**

If any atom group that does not match ``select`` is selected for RMSD calculation in ``groupselections``, we get a ValueError because the size of the weights array does not match the selected atom group. Also, the masses selected are those for the reference atoms, not the mobile ones.

https://github.com/MDAnalysis/mdanalysis/blob/3e8ea8e0744f3163ddbf7b4f5f66874dd26796a7/package/MDAnalysis/analysis/rms.py#L557-L558

This is because `self.weights` is precomputed in `self._prepare()` as a single array. Instead, following @orbeckst's suggestion below, it should always be an iterable of arrays and raise an error when `groupselections` is provided, but only a single array of `weights` is provided.

eg.
```python
        if self.weights == 'mass':
            self.weights = [self.mobile_atoms.masses]
            for atoms in self._groupselections_atoms:
                self.weights.append(atoms['mobile'].masses)
        if self.weights is not None:
            if len(self.weights) != len(self.groupselections) + 1:
                raise ValueError('... message about weights!...')
```

**Code to reproduce the behavior**

Show us how to reproduce the failiure. If you can, use trajectory files from the test data.

```python
>>> import MDAnalysis as mda
>>> from MDAnalysis.tests.datafiles import PSF, DCD, CRD
>>> import MDAnalysis.analysis.rms as rms
>>> mda.__version__
'0.20.1'
>>> u = mda.Universe(PSF, DCD)  # closed AdK (PDB ID: 1AKE)
>>> R = rms.RMSD(u,  # universe to align
...              u,  # reference universe or atomgroup
...              select='backbone',  # group to superimpose and calculate RMSD
...              groupselections=["all"],  # groups for RMSD
...              weights='mass')  # weights for each atom
>>> R.run()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "MDAnalysis/analysis/base.py", line 197, in run
    self._single_frame()
  File "MDAnalysis/analysis/rms.py", line 638, in _single_frame
    center=False, superposition=False)
  File "MDAnalysis/analysis/rms.py", line 247, in rmsd
    raise ValueError('weights must have same length as a and b')
ValueError: weights must have same length as a and b
```

Note: ``weights='mass'`` does work when only ``select`` is passed and there are no ``groupselections``.

**Currently version of MDAnalysis**

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
0.20.1
- Which version of Python (`python -V`)?
3.7.x
- Which operating system?
MacOS

I get that this becomes tricky when an array of weights is passed, but when ``RMSD`` is given "mass" as a string it should be fairly straightforward to apply it to the ``groupselections`` as well.

Also, I think it would be more flexible (if less convenient) to require that a ``weights`` array be the same shape as the entire universe, so that all selections can be weighted, rather than passing an array the same shape as ``select`` and failing for ``groupselections`` that necessarily will have different atoms.

**Comment by orbeckst** [comment]
I am labeling this as a bug because what you describe is a very sensible user expectation and the resulting error is confusing. However, the `groupselections` feature is still labeled as experimental, so I am not surprised that not everything works seamlessly. I think it was never properly defined how these things _should_ work. Additionally, it's not even really documented what kind of output `groupselections` produces – one has to deduce from the [example](https://www.mdanalysis.org/mdanalysis/documentation_pages/analysis/rms.html#calculating-rmsd-for-multiple-domains) that the additional RMSDs are added as columns to the output.

My current thoughts:

- At a minimum `RMSD` should fail with a clear error message for this case (and we need to document the limitation). 
- Ideally, it works "as expected" when `weights == "mass" `is provided in that the appropriate mass arrays are generated for each group selection.
- Custom mass arrays are pretty advanced so I think it would be ok to require them to be
   -  a 1D array of the same length as the AtomGroup resulting from `select` for `groupselections == None` (the current case)
   - a list of 1D arrays that match the length of the fitting selection (from `select` or the `atomgroup` itself) and all the AGs resulting from `groupselection`. I.e., the `weights` would be a list of length 1 + N where N is `len(groupselections)`.

**Comment by yuxuanzhuang** [comment]
Hi! I have linked a PR for review. The new rmsd function now supports:
![showcase](https://user-images.githubusercontent.com/24636656/75929874-bdfcd400-5e71-11ea-82a2-7de8e1f99bfb.png)

**Comment by orbeckst** [comment]
@yuxuanzhuang see comments on your PR #2577 

By the way, your PR looks reasonably complicated and will likely take a while to review and finalize. I would suggest that in parallel you put up a smaller PR that just addresses
> Also, the masses selected are those for the reference atoms, not the mobile ones.

This will be easy to review and to test.

**Comment by yuxuanzhuang** [comment]
@orbeckst I added a fix and a short test to the PR #2577 for that. Should I also put up a new fix PR for the original code?
Generally, my solution is to expand the input `weights` from **'mass'**, **'None'**, **'1D array'** or a list of them to reconstructed 1-D weight arrays and save them in `weights_ref`, `weights_select`,`weights_groupselection` in `_prepare` stage.
